app/actions/local/reactions.test.ts

```diff
@@ -0,0 +1,111 @@
+
+    beforeEach(async () => {
+        jest.clearAllMocks();
+
+        await DatabaseManager.init([serverUrl]);
+        operator = DatabaseManager.serverDatabases[serverUrl]!.operator;
+
+        (recentReactionsQueries.getRecentReactions as jest.Mock).mockResolvedValue([]);
+        (emojiHelpers.getEmojiFirstAlias as jest.Mock).mockImplementation((emoji) => emoji);
+
+        operator.handleSystem = jest.fn();
+    });
+
+    afterEach(async () => {
+        await DatabaseManager.destroyServerDatabase(serverUrl);
+    });
+
+    it('should return an empty array if emojiNames is empty', async () => {
+        const result = await addRecentReaction(serverUrl, []);
+        expect(result).toEqual([]);
+    });
+
+    it('should add new emoji to the beginning of the list', async () => {
+        const emojiNames = [':air:', ':fire:'];
+        await addRecentReaction(serverUrl, emojiNames);
+
+        expect(operator.handleSystem).toHaveBeenCalledWith({
+            systems: [{
+                id: 'recentReactions',
+                value: JSON.stringify([':fire:', ':air:']),
+            }],
+            prepareRecordsOnly: false,
+        });
+    });
+
+    it('should move existing emoji to the beginning of the list', async () => {
+        (recentReactionsQueries.getRecentReactions as jest.Mock).mockResolvedValue([':fire:', 'water']);
+        const emojiNames = [':air:', ':fire:'];
+        await addRecentReaction(serverUrl, emojiNames);
+
+        expect(operator.handleSystem).toHaveBeenCalledWith({
+            systems: [{
+                id: 'recentReactions',
+                value: JSON.stringify([':fire:', ':air:', 'water']),
+            }],
+            prepareRecordsOnly: false,
+        });
+    });
+
+    it('should limit the list to MAXIMUM_RECENT_EMOJI', async () => {
+        const longEmojiList = Array.from({length: 40}, (_, i) => `emoji${i}`);
+        (recentReactionsQueries.getRecentReactions as jest.Mock).mockResolvedValue(longEmojiList);
+        await addRecentReaction(serverUrl, ['newEmoji']);
+
+        const handleSystemCall = (operator.handleSystem as jest.Mock).mock.calls[0][0];
+        const savedEmojis = JSON.parse(handleSystemCall.systems[0].value);
+        expect(savedEmojis.length).toBe(27);
+        expect(savedEmojis[0]).toBe('newEmoji');
+    });
+
+    it('should use getEmojiFirstAlias for each emoji', async () => {
+        const emojiNames = [':air:', ':fire:'];
+        await addRecentReaction(serverUrl, emojiNames);
+
+        expect(emojiHelpers.getEmojiFirstAlias).toHaveBeenCalledTimes(2);
+        expect(emojiHelpers.getEmojiFirstAlias).toHaveBeenCalledWith(':air:');
+        expect(emojiHelpers.getEmojiFirstAlias).toHaveBeenCalledWith(':fire:');
+    });
+
+    it('should handle errors and log them', async () => {
+        const failedError = new Error('Test error');
+        (recentReactionsQueries.getRecentReactions as jest.Mock).mockRejectedValue(failedError);
```

Comment:
The function that may throw in this component is `getServerDatabaseAndOperator`. The rest of the functions shouldn't throw.

A quick way to make this throw is to use a server url that is not registered.

Reply:
Got it.
I'll pass unregisterd server url.